### PR TITLE
Add synccontroller for reliable message delivery

### PR DESCRIPTION
**Pull request by fisherxu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test

/kind feature

**What this PR does / why we need it**:
Add the synccontroller to resend the failed persisted objects to edgenode.
1. Compare the  objects that have been persisted to the edge with the objects in K8s  and generate update and delete events to the edge.
2. Compare the objects in K8s with the objects that have been persisted to the edge,
   If the objects fail to persisted to the edge for the first time, it will be recreated here.
   Just focus on the pod, service, endpoint, because if the pod is passed down, and the configmap and secret cannot be found and which will be queried to the cloud.

**Which issue(s) this PR fixes**:

Ref: #1371 
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add synccontroller for reliable message delivery
```

### CODE REVIEW

**Comment by fisherxu** [comment]
It's ready to review now :) cc @kevin-wangzefeng @kadisi

File: cloud/pkg/synccontroller/createfailedobject.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+	set := labels.Set{edgemgr.NodeRoleKey: edgemgr.NodeRoleValue}
+	selector := labels.SelectorFromSet(set)
+	allEdgeNodes, err := sctl.nodeLister.List(selector)
+	if err != nil {
+		klog.Errorf("Filed to list all the edge nodes: %v", err)
+	}
+
+	for _, pod := range allPods {
+		if isFromEdgeNode(allEdgeNodes, pod.Spec.NodeName) {
```
Comment (kadisi) [inline cloud/pkg/synccontroller/createfailedobject.go]:
```
if !isFromEdgeNode(allEdgeNodes, pod.Spec.NodeName) {
  continue
}
```
?

File: cloud/pkg/synccontroller/synccontroller.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+}
+
+// Group of controller
+func (sctl *SyncController) Group() string {
+	return SyncControllerModuleGroup
+}
+
+// Group of controller
+func (sctl *SyncController) Enable() bool {
+	return true
```
Comment (kadisi) [inline cloud/pkg/synccontroller/synccontroller.go]: add `Enable` field for `SyncController  struct` and then use sctl.Enable.

File: cloud/pkg/synccontroller/createfailedobject.go (hunk 1 above)
Comment (fisherxu) [reply cloud/pkg/synccontroller/createfailedobject.go]: Done.

File: cloud/pkg/synccontroller/synccontroller.go (hunk 2 above)
Comment (fisherxu) [reply cloud/pkg/synccontroller/synccontroller.go]: Done.

**Pull request merged**